### PR TITLE
Handle `"true"` being passed for the `email_verified` OIDC claim

### LINKED ISSUE #14851: email_verified in OIDC claims doesn't allow String values
Expected behavior:
When the OIDC IdP returns the `email_verified` claim as string Teleport should convert that into a boolean so it's able to verify the claim.
```
{
    "email": my@mail.com,
    "email_verified": "true",
    "exp": 1658764924,
    "aud": "xxx",
    "iat": xxx,
    "iss": https://myidp/path/to/issuer
}
```

Current behavior:
When the `email_verified` claim is returned as string - this was tested with the claim being set to all caps "TRUE" and lower caps "true" - then Teleport OIDC login will fail with the error message `OIDC provider did not verify email.\n\tunable to parse oidc claim: \"email_verified\", must be a string or bool`.

If the `email_verified` claim is set to a boolean, e.g.:

```
{
    "email": my@mail.com,
    "email_verified": true,
    "exp": 1658764924,
    "aud": "xxx",
    "iat": xxx,
    "iss": https://myidp/path/to/issuer
}
```
The authentication seems to work (tested with a different IdP / setup since our IdP doesn't support booleans for the email_verified claim).

Bug details:
- Teleport version: v10.0.2
- Recreation steps: see current behavior / expected behavior.
- Debug logs

> ERROR REPORT:
Original Error: *trace.BadParameterError unable to parse oidc claim: &#34;email_verified&#34;, must be a string or bool
Stack Trace:
Caught:
                /go/src/github.com/gravitational/teleport/lib/httplib/httplib.go:167 github.com/gravitational/teleport/lib/httplib.ConvertResponse
                /go/src/github.com/gravitational/teleport/lib/auth/clt.go:250 github.com/gravitational/teleport/lib/auth.(*Client).PostJSON
                /go/src/github.com/gravitational/teleport/lib/auth/clt.go:976 github.com/gravitational/teleport/lib/auth.(*Client).ValidateOIDCAuthCallback
                /go/src/github.com/gravitational/teleport/lib/web/apiserver.go:1249 github.com/gravitational/teleport/lib/web.(*Handler).oidcCallback
                /go/src/github.com/gravitational/teleport/lib/web/apiserver.go:2665 github.com/gravitational/teleport/lib/web.(*Handler).WithMetaRedirect.func1
                /tmp/gomodcache/github.com/gravitational/httprouter@v1.3.1-0.20220408074523-c876c5e705a5/router.go:399 github.com/julienschmidt/httprouter.(*Router).ServeHTTP
                /opt/go/src/net/http/server.go:2127 net/http.StripPrefix.func1
                /opt/go/src/net/http/server.go:2084 net/http.HandlerFunc.ServeHTTP
                /go/src/github.com/gravitational/teleport/lib/web/apiserver.go:465 github.com/gravitational/teleport/lib/web.NewHandler.func1
                /opt/go/src/net/http/server.go:2084 net/http.HandlerFunc.ServeHTTP
                /tmp/gomodcache/github.com/gravitational/httprouter@v1.3.1-0.20220408074523-c876c5e705a5/router.go:460 github.com/julienschmidt/httprouter.(*Router).ServeHTTP
                /go/src/github.com/gravitational/teleport/lib/web/apiserver.go:217 github.com/gravitational/teleport/lib/web.(*APIHandler).ServeHTTP
                /tmp/gomodcache/github.com/gravitational/oxy@v0.0.0-20211213172937-a1ba0900a4c9/ratelimit/tokenlimiter.go:118 github.com/gravitational/oxy/ratelimit.(*TokenLimiter).ServeHTTP
                /tmp/gomodcache/github.com/gravitational/oxy@v0.0.0-20211213172937-a1ba0900a4c9/connlimit/connlimit.go:75 github.com/gravitational/oxy/connlimit.(*ConnLimiter).ServeHTTP
                /opt/go/src/net/http/server.go:2916 net/http.serverHandler.ServeHTTP
                /opt/go/src/net/http/server.go:3523 net/http.initALPNRequest.ServeHTTP
                /opt/go/src/net/http/h2_bundle.go:5903 net/http.(*http2serverConn).runHandler
                /opt/go/src/runtime/asm_amd64.s:1571 runtime.goexit
User Message: OIDC provider did not verify email. unable to parse oidc claim: &#34;email_verified&#34;, must be a string or bool
] web/apiserver.go:1251

Error seems to happen here: https://github.com/gravitational/teleport/blob/master/lib/auth/oidc.go#L381 when https://github.com/gravitational/teleport/blob/master/lib/auth/oidc.go#L393 is called.

### PR DESCRIPTION
**Pull request by ryanclark**
This allows for the string `"true"` to be given for the `email_verified` OIDC claim.

Previously, it would just allow `"false"`, which would cause the error `email_verified must be a string or a bool`.

I've updated it to allow for any string, checking if it's equal to `true` or `false`, and returning a bad parameter error if it's set to anything else.

Fixes #14851

### CODE REVIEW

File: lib/auth/oidc.go [hunk 1]
```diff
@@ -369,8 +369,15 @@ func checkEmailVerifiedClaim(claims jose.Claims) error {
...
-		return unverifiedErr
+	if hasEmailVerifiedClaim {
+		if emailVerified == "false" {
+			return unverifiedErr
+		}
+		if emailVerified == "true" {
+			return nil
+		}
+
+		return trace.BadParameter("unable to parse oidc claim: %q, must be set to either 'true' or 'false'", claimName)
```
Comment (zmb3) [inline lib/auth/oidc.go:380]: Maybe include the actual value of `emailVerified` as well to help troubleshooting.

Comment (ryanclark) [reply lib/auth/oidc.go:380]: Good shout

File: lib/auth/oidc.go [hunk 2]
```diff
@@ -369,8 +369,15 @@ func checkEmailVerifiedClaim(claims jose.Claims) error {
 	unverifiedErr := trace.AccessDenied("email not verified by OIDC provider")
 
 	emailVerified, hasEmailVerifiedClaim, _ := claims.StringClaim(claimName)
-	if hasEmailVerifiedClaim && emailVerified == "false" {
-		return unverifiedErr
+	if hasEmailVerifiedClaim {
+		if emailVerified == "false" {
```
Comment (Tener) [inline lib/auth/oidc.go:373]: Perhaps we can allow more formats? If so, https://pkg.go.dev/strconv#ParseBool might be a good choice.

Comment (zmb3) [reply lib/auth/oidc.go:373]: No need to support more formats. Technically, even "true" is out of spec, but one or two IDPs we know of get it wrong.

**Review by zmb3 — APPROVED** [review]

File: lib/auth/oidc.go (hunk 2 above)
Comment (r0mant) [inline lib/auth/oidc.go:373]: Should we use `strings.ToLower` here and below just in case?

**Review by r0mant — APPROVED** [review]

File: lib/auth/oidc.go (hunk 2 above)
Comment (zmb3) [reply lib/auth/oidc.go:373]: We decided not to. Want to be as strict as possible since we're already out of spec here.

**Pull request merged**